src/analysisd/decoders/dbsync.c

```diff
@@ -92,6 +92,13 @@ static void dispatch_answer(dbsync_context_t * ctx, const char * result) {
     char * query;
     os_malloc(OS_MAXSTR, query);
 
+    ComponentType component_type = validate_component(ctx->component);
```

Comment:
We can remove this and centralize the component check at `DispatchDBSync()`

Reply:
Moved!